synapse/swarm/syntax.py

```diff
@@ -233,7 +232,11 @@ def parse_ques(text,off=0,trim=True):
             ques['valu'],off = parse_literal(text,off+1,trim=True)
             break
 
-        # TODO: handle "by" syntax
+        if text[off] == '*':
+            ques['cmp'],off = nom(text,off+1,ascii_lowercase,trim=True)
+            assert text[off] == '='
+            ques['valu'],off = parse_list(text,off+1,trim=True)
```

Comment:
Not every "by" handler is going to expect a list argument.  This should just be parse_literal and we'll let the individual "by" index callbacks raise if they see fit.